bom/runtime/pom.xml

```diff
@@ -2023,6 +2023,13 @@
                 <groupId>io.jaegertracing</groupId>
                 <artifactId>jaeger-thrift</artifactId>
                 <version>${jaeger.version}</version>
+                <exclusions>
+                    <!-- fixme, jaeger-thrift 3.43.3 pulls in a banned version of annotation-api, so we exclude for now-->
+                    <exclusion>
+                        <groupId>javax.annotation</groupId>
+                        <artifactId>javax.annotation-api</artifactId>
+                    </exclusion>
```

Comment:
Usually, we put the exclusion here: https://github.com/quarkusio/quarkus/blob/master/extensions/jaeger/runtime/pom.xml#L28 and we also add the corresponding jakarta dependency as that's what we want to use now.

Reply:
I've moved the exclusion, but it doesn't really make sense to force the jakarta dependency as it is a transitive dependency sometime inside the jaeger-thrift dep tree.

Reply:
Not sure I follow you.

If it requires the annotation-api jar then we'd better have it around so we would need the jakarta one instead of the javax one as that's the ones we use.

Why do you think it wouldn't be necessary?

Reply:
Not a big issue and I can also add it if you prefer. 

My reasoning is that if the dependency changes/is removed inside jaeger-thrift, then it will still be required by this pom - and it won't really be discovered that it can be removed. Putting this kind of transitive dependencies messes with the dependencies graph and causes the number of dependencies to grow larger than it needs to over time. 

For the dependency in question, it probably won't make a difference, the jakarta api-annotation lib is pull in through a lot of other places, so it isn't likely to be removed anytime soon.

Reply:
So I merged it. But your reasoning is flawed as soon as the library itself really needs the dependency.

Not adding it only works if it's not necessary or if something else brings it. If that something else stops bringing it, something will get broken.

Anyway, not a big deal in this case, I agree.

Reply:
Well, by that logic, lets add it anyways. 

New pull request here:
https://github.com/quarkusio/quarkus/pull/10773